h/accounts/models.py

```diff
@@ -132,6 +133,11 @@ def is_activated(self):
 
         return False
 
+    def activate(self, request):
+        """Activate the user by deleting any activation they have."""
+        request.db.delete(self.activation)
+        request.registry.notify(events.ActivationEvent(request, self))
```

Comment:
I just got myself caught in an unfortunate cycle, when trying to anticipate what would pass review. I can see pros and cons of all three solutions (including clear cons of putting it on the model as I decided to) but to me they seem equally weighed. Initially I had it duplicated, but thought that wouldn't get through review (turns out it would have). Next I thought of a logic module, but guessed that adding a logic module just for one function wouldn't get passed review (turns out, it would have). So I ended up adding it to the model class, thinking that had the best chance of making it through the review (turns out, it didn't!).

I'll put it back to duplication...